core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -264,5 +253,35 @@ public void close() {
           "File groups offered after service was closed, "
               + "they were not successfully committed.");
     }
+
+    private void commitReadyCommitGroups() {
+      if (canCreateCommitGroup()) {
+        synchronized (completedRewrites) {
+          if (canCreateCommitGroup()) {
+            String inProgressCommitToken = UUID.randomUUID().toString();
+            Set<RewriteFileGroup> batch = Sets.newHashSetWithExpectedSize(rewritesPerCommit);
+            for (int i = 0; i < rewritesPerCommit && !completedRewrites.isEmpty(); i++) {
+              batch.add(completedRewrites.poll());
+            }
```

Comment:
I think the synchronization needs to end before you start the commit or else they will happen one at a time correct?

Reply:
Synchronize (completeRewrites) {
   create batch
}
try commit batch

Reply:
Otherwise if two offers each need to commit, we have to wait for one to finish before the next can start

Reply:
+1, makes sense, making the changes